huaweicloud/services/vpn/resource_huaweicloud_vpn_gateway.go

```diff
@@ -121,21 +121,51 @@ func ResourceGateway() *schema.Resource {
 				Computed:    true,
 				Description: `The enterprise router ID to attach with to VPN gateway.`,
 			},
+			"ha_mode": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				Computed:     true,
+				Description:  `The HA mode of the VPN gateway.`,
+				ValidateFunc: validation.StringInSlice([]string{"active-active", "active-standby"}, false),
```

Comment:
the ha_mode is fixed to "active-standby" when "master_eip" and "slave_eip" are specified
```suggestion
				ValidateFunc: validation.StringInSlice([]string{"active-active", "active-standby"}, false),
				ConflictsWith: []string{"master_eip", "slave_eip"},
```